MetaMask/metamask-extension
# ConfirmPage: update UI for setApproveForAll transactions to better warn users of the allowances

## Explanation

In https://github.com/MetaMask/metamask-extension/pull/15496, @cygaar proposed improving the warning message on the ConfirmPage for `setApprovalForAll` transaction types and gathered feedback to support this improvement.

Based on an internal dialog [here](https://consensys.slack.com/archives/GTQAGKY5V/p1659966850991299?thread_ts=1659958204.365809&cid=GTQAGKY5V), we will be updating the "Confirm" button to red, and including an error type Dialog to better warn users that they are interacting with a `setApprovalForAll` transaction. This is a temporary improvement while a redesign is on the way. More details are explained in https://github.com/MetaMask/metamask-extension/pull/15496 and the [internal dialog](https://consensys.slack.com/archives/GTQAGKY5V/p1659966850991299?thread_ts=1659958204.365809&cid=GTQAGKY5V).

## More Information

See: https://github.com/MetaMask/metamask-extension/pull/15496
See: [Internal Dialog](https://consensys.slack.com/archives/GTQAGKY5V/p1659966850991299?thread_ts=1659958204.365809&cid=GTQAGKY5V)

## Screenshots/Screencaps
### Before
<img width="468" alt="Screenshot 2022-08-10 at 3 17 20 AM" src="https://user-images.githubusercontent.com/20778143/183743004-23ee911a-8393-4e83-a3c7-0822171905dc.png">

### After
<img width="526" alt="Screenshot 2022-08-10 at 6 06 42 PM" src="https://user-images.githubusercontent.com/20778143/183882919-ca3f15b4-dd5e-4751-a9bb-07d91567fa30.png">
<img width="406" alt="Screenshot 2022-08-10 at 6 07 04 PM" src="https://user-images.githubusercontent.com/20778143/183882944-4ee92462-b118-48f5-91f1-604bc60876a0.png">


## Manual Testing Steps

- send a `setApprovalForAll` transaction and observe a red, danger-primary "Confirm" button
- send any other transaction and observe a blue, primary "Confirm" button

### + If there are functional changes:
